cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractProgramController.java

```diff
@@ -141,11 +181,11 @@ public void run() {
    */
   protected void complete() {
     if (!state.compareAndSet(State.ALIVE, State.COMPLETED)) {
-      LOG.debug("Cannot transit to COMPLETED state from {} state: {} {}", state.get(), programName, runId);
+      LOG.debug("Cannot transit to COMPLETED state from {} state: {} {}", state.get(), name);
       return;
     }
-    LOG.debug("Program stopped: {} {}", programName, runId);
-    executor(State.COMPLETED).execute(new Runnable() {
+    LOG.debug("Program controller stopped: {}", name);
```

Comment:
I think either all lifecycle transitions should log, or none should log. Here, it seems that only complete() logs.